### PR TITLE
refactor: Add logging to mobile IAP checkout/ API

### PR DESCRIPTION
**Pull request by moeez96**
# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

### TODO: Add auto_retry_wrong_env_request to environment variables
auto_retry_wrong_env_request = True retries the payment validation on Apple sandbox servers in case the receipt belongs to a different environment on Apple.
#### Quoting [Apple docs](https://developer.apple.com/documentation/appstorereceipts/verifyreceipt):
`As a best practice, always call the production URL https://buy.itunes.apple.com/verifyReceipt first and proceed to verify with the sandbox URL if you receive a 21007 [status](https://developer.apple.com/documentation/appstorereceipts/status) code. Following this approach ensures that you don’t have to switch between URLs while your app is in testing, in review by App Review, or live in the App Store.`

More on [receipt validation status](https://developer.apple.com/documentation/appstorereceipts/status).

Hence the variable is not added to environment variables and set to `True` instead.

### TODO: Add check for products_in_basket_already_purchased and TODO: Add logging for api/user_info after reading from request obj

In the checkout/ API case, the request is received in `iap.api.v1.views. MobileCheckoutView` as Rest Framework request object, while the second view called i.e. `extensions.api.v2.views.CheckoutView` expects a Django HttpRequest. Which is forwarded by accessing the protected attribute `request._request`.
Reading the data stream from the POST payload causing the error `You cannot access body after reading from request's data stream` in the second view while accessing the payload resulting in a 500.

#### Link to error: https://onenr.io/0Zw08Db90wv

Hence logging has been added for exception without accessing payload.

## Other information

Jira link: https://2u-internal.atlassian.net/browse/LEARNER-9347

### CODE REVIEW

File: ecommerce/extensions/iap/api/v1/ios_validator.py [hunk 1]
```diff
@@ -12,10 +12,9 @@ def validate(self, receipt, configuration):
         Apple for the mentioned productId.
         """
         bundle_id = configuration.get('ios_bundle_id')
-        # If True, automatically query sandbox endpoint if validation fails on production endpoint
-        # TODO: Add auto_retry_wrong_env_request to environment variables
-        auto_retry_wrong_env_request = True
-        validator = AppStoreValidator(bundle_id, auto_retry_wrong_env_request=auto_retry_wrong_env_request)
+        # auto_retry_wrong_env_request = True automatically queries sandbox endpoint if
+        # validation fails on production endpoint
+        validator = AppStoreValidator(bundle_id, auto_retry_wrong_env_request=True)
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/api/v1/ios_validator.py:17]: We should move this to environment flag.

**Review by jawad-khan — APPROVED** [review]

File: ecommerce/extensions/iap/api/v1/ios_validator.py (hunk 1 above)
Comment (moeez96) [reply ecommerce/extensions/iap/api/v1/ios_validator.py:17]: Discussed on PM this is not needed and we are good to go. Resolving.

**Comment by moeez96** [comment]
Rebased to master branch

**Review by grmartin — APPROVED** [review]
LGTM

**Pull request merged**